### ISSUE TITLE
Add `ignore: ["with-var-inside"]` to `color-function-notation`

### ISSUE THREAD

**Issue by boris-petrov** [stylelint/stylelint#6620]
### What steps are needed to reproduce the bug?

```css
div {
  background-color: rgba(var(--bs-primary-rgb), 0.4);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "ignoreFiles": ["**/*"],
  "overrides": [
    {
      "files": ["**/*.css"],
      "extends": ["stylelint-config-standard", "stylelint-config-prettier"],
      "rules": {}
    },
    {
      "files": ["**/*.scss"],
      "extends": ["stylelint-config-standard-scss", "stylelint-config-prettier"],
      "rules": {}
    }
  ],
  "rules": {}
}
```

### How did you run Stylelint?

`stylelint app/styles`

### Which version of Stylelint are you using?

14.16.1

### What did you expect to happen?

No error/warning showing since the syntax is correct.

### What actually happened?

An error is reported.

### Proposal to fix the bug

Maybe related to https://github.com/stylelint/stylelint/issues/5671.

**Comment by jeddy3** [comment]
@boris-petrov Thanks for opening the issue and using the template.

I believe the rule is correctly flagging the pattern ([see demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIoEsBuACYAdAdu9ARgIZgDWA5gE4QCuOUAtJADYSXzqXnEAUqRlHgwYEAzgwAOlZAFsBATwZcCASgA06AAwA6ACwqA3LgC+uEGpCj5OAC5EAHgnCjR58BBwAzZOSfY86FgglDTMMKJBHP74+EFEODgQdjbIHgyJDHSkiQDuOJHoNiEwargxgSBENkqhMOkQmTjZEHkFRTQlZTFBBKxk9QwwMhI28m3FpQGx7qyUA8g4-MzIjAAWMPbjHZPlQZAyMjC2A0MjYwiFE13TYDSiNhAyktQSMJSjAzLILgvkDPxzTx0MApDxbTpTCqwMDMARVVI4ER9UgDKA0CTLMBVOq3e6PZ4QV7vZDhcE7bpIGAwuGgxG9CD9DJojHILE2OpSQlvFKki4AbWuMXaEPK+GioumPkSlBgBQFkIlew8oipNBSqDqzMx2PEOWQNlWDBQnk8byO1SWHQiIEF5QAurb0KZIXbydNobDKPC0vTGQ1RKs2Ab4oxOUSPhANZRpLBrRxhW6Kp4PNVPEQvsxFEz0dr2el07z41dIUFk8c0xmsw0vj8cH9yEc3qyGBXkJmGKQYPIcmwoGTBaXgbSmERmGABnRRBISDBGFyvQ9KP2SyAgTgQQiGMscDABAwqEQUOaBokcPcQwJGCgZRuwRcEwPV0PNxksrl8vfi7sQJ35J4vYcSIMii2YsmydQqmEIJsHGlzbI+v7-gWRpUp63qIhksgSEG8Q2Mu36HCgRAtruNg0DK+aAa+TTvvhFI4AWjBUCsDACLu4iYYso4rHR0xMuEYBHCgdYMFOVLIN4YD6ucRbwSuoG5nUVQMFhQY1GEsEPvJDRaqy2KiTAUGLppX70Q0pwfKItCUIJvEVBkGxFMRKpfCwyp2UEnFLCxUC0L0EGwgGTCPIctgmXJ35edxoYQKI+qbspqnvOpsqfhFZkqdGMDkKEe45Ks+rhNOtlpSKFJhtyVaNM0rSlYmQSQVSi6SCqNC+UwgUcQ0b4tB+slldMjXQXMEite1hlDMe1E1X1cEDRUQ3NaMrwTjRvVyo6WkSuKiogFKbCpRw8oSt+uIPE8E2hTY1qOvgDoKk6gquo+9zSCJGQ7jk26HXN9UgHQ+qrTNHmVMwEirMRlp1IkyQIhtD1BESgm2EQDZBImYq3RUGyCSMAAKLzciSsHHSd0yEiQ0no1j0zeMwzAMBTUmjNTD3fp4rDQIzxVU+YNMLQ8Ejc5TLN82zFKvRAnbC8z5xY-dErOvaf3KSEYSDMMHzfSIMDJjK8O7aOOREPI1oY5gWNBDjMAjAbZMVL6pAaeIRCeOycyiMhDGAY7zus-bSbIJQ9zpOE7J9ja4vPeLQT7fr-L83EbtvMFBzmkE8uOkrMTR9+qu1JROJECqBRBKwORvP7FL5+r3sQcVSnJ0uFxxMwxum6Jvzq99VfTDX0MNBq9BsM8uvIJsdWPv3BmuRArAYTAX0LI37ul6D7dm4+jsdbFvwiF6gka2cW7Lzres-Z5MBRr3DvIjvcUiQQB-Q4v2uu6vLfrybm8rtvMK74-Z+ocl47jPgdNeRtv4MBkKEFIJ8dw3x6Hff+D8-hPxnKJBuYD44cFbhvTudZu7L0QSAbeXIcB73QYfT6b8m4QLblAmBzA4E9zFt+MhrwKGAIwVODB79K6f0gR3VBRCEFsIpBwo4lCgG8MPgQXW4DBEMNNiQtygJnxpBhuhNeMhoBvA-H9NRDB1j2BHCXT+5cBGGLniPExW4jjkANGvAMQZVEhWPBZRQ2t5HnztnnZRP9tqWxANbW2Cdxa0yDiHHc9xZwZyjubSU5BpQ-VJmTBqowwjbmqPsOQ9AbpRyzk9axadjj5UKrIuoCw4qwHoRvVRdxzoEnDIoTx8C6g+MUUdR0eDv430xjHEJ9hcZ4XCQHJO7smCNPxBVd45xEkUm8MHaoMTw7xLJrnE6sdkldPQGkrZlQm6pyuv00UsczwrAgl3Oo3jkTrJOgrUU2c7qGOmU8QiyBiLThsO7Wa+zpgAD0eB8mIgALztCChgoLNAMAAJx2gAFQqCEJC6FcK7QAGoVAIoACSnOAM89AmybhvOaZVSQVRfl+IpEC1FEKwUwvhUilFDL0VYtxfiwlxKoSoRpJuYgIlKmsToZ-Hc19xHul5Qufl8Q-hCs6TgioQjAkUg9Hyn0d8XKshsQvEBK8BG4K-h3JhLDiESp5dSaVGrgIz21fPTBfCRWGuVQQ8gojUp-TVVaukmqhh2rSPKhRirL7is9VK9CQF+giJuafOQpivXoVggARjDZaiN28nJtj3lqtys1ekqPNUEBNw5t4ZBlGieguEtweHIBDegZLiSFl+o+Ytm5c3ANoR-Z1AToGwOQO0khra0jtqFfw5u3b8HRoHYWykabhwjqwQqi+IAxVWJbeG4cbTvFBtST0o1Kryg7TOUMkZ1LDZHKHQYxOq4okrLDnEyOGyFnbJSWe49Y7jnp3NbtaplzXXusjaQe5EpHnlC5X9MsqZ0xtizMhAAjjQJITb826nWBRG87j6APogxoxEuTiI0NPmOupjC+3Tpw+uedIVnJYOI0oyd1zyOPjXLePD1GHVyJ3WvVdzcKOsegQ4I+Wtl6wU0Hx4cdczHLssbx5juGKUylsOsFUHFX6n1-bU+jpHmH9tYeJzc05FMGnCOEDjVSLmacNTx-9MaxH6bSORBmCGkOwRQyQljw5ynslHU6pVASSGrA6ChKgMAB6wzvM6whHrHxJRWUkbRn9HMkIWLAFGtICgACY-qIQAhBCaw16jhbzSAJG5pUZ1DoM7BgeoDQLA7F2HslA5w4HbI7EhOWCwcWQt8qlYzj20rBfSqFjLEXIoYKikb7K8XfswOBx8YQ6wGihW8Bo1F9QgzjUJrxImCgpsfJt7WC3HGrF2xlsT+3ZyfJIlUciOIdVmeFV2vz9SZ0fOIqaG7FEF0YKXdxq+a6Vxveu2RCikmsTmMNTJkhQOPsg+hl7QeQkR5SDHhPfqf0YekVuwp80ynTNCo08unj0PLvvax+h2VdR5xVBHj5p7bnXuk+B9jr0dYqdEhpx7RdXGtMFox0zhDbxtsh3wx2ojvmUO9p00xwHAuOiUGFzk9jdODXPagVOvTF2iIMEFwrk+Ivlfc98aK-7sntJbZbLe6d6PHwOTntVgq3mG4gwyDWVBgx60QE8KJay1C1NiJt-JGgMh5FzDCIePeoKVskb57b4PoeBOmJR1JOKEX0C6D+jgePKcMiZu3H8KP1BwrzURoTOZUm15Q5nbMj4GQh6+RGjKbwaPm0rjVnULdp9ft9f8fgk1umzXmyPd+UJozukRMDss0OsSI7y2fXtHZir-nvqObkr9mcHpzZXItEelLpAEDVB0g+nZroPcJ390NL18vNT38gA-eZqfNRV+O+ypuSE77mLf+-7O3ic4e93qzN-GdD-ViH5ffQ-HXRDbzGPA9Baa-EeakFwClH5fRN9IIAbKFIbNFJlMbCbNlbFabIfLfb8EA-YAgBYP-Z-GA9-eAuYMgigp-I3XZBnP6EA2MZGKAKteghiZqD6ANZ3SebfWglSUMSlVAnvGlYFQbPAnAllYbfAjlGbAlYpK-IyEeZYA3A4Ajf3fVF-Fg1QpqdQ74JXLQ--HnQAy-IQtQuYTbTvGJAoc7KwwwuYOvJHRvVHEGEAmGQrBLQ1fYDEDYGg6wlqGANqBoRA8QcHaTFoAHEg4Q0aUI9qCInHJTEzcQAnCzInIA1g+IsacyMIK6CvCxGIs3OI4IhIsIwYAo48dtLRdLT+XyGgfyII5whgZaIuCHCoKvVgwMd4OtUMMvWvBoctOgTg44KGYvVgooPeZzaAhovyMIEhAGHJYuaIiuUoikKGerbsXsIoyHEokhLYjQkwuQMXUBOjCdbTU1OzR8I44w45GjR1enfdGzGXb8O4zQ04wNY3Cw2IzY0cILY4xPC3b6UTP6LY1w4edw5vN9LaUUYfCkOOGAAANQBKbWX3YQgAng30Viei6GdGMBAGMCAA)) as the following is legacy notation:

```css
div {
  background-color: rgba(var(--bs-primary-rgb), 0.4);
}
```

Modern notation is:

```css
div {
  background-color: rgb(var(--bs-primary-rgb) / 0.4);
}
```

The standard configs enforce modern notation as they encourage users to write forward-facing CSS.

**Comment by boris-petrov** [comment]
@jeddy3 thanks for the discussion! However, I believe it depends on the contents of the variable. Examples:

```
Case A:
--bs-primary-rgb: 255, 128, 64;

color: rgba(var(--bs-primary-rgb), 0.5); /* this is valid */
color: rgba(var(--bs-primary-rgb) / 0.5); /* this is not valid */

Case B:
--bs-primary-rgb: 255 128 64;

color: rgba(var(--bs-primary-rgb), 0.5); /* this is not valid */
color: rgba(var(--bs-primary-rgb) / 0.5); /* this is valid */
```

**Comment by Mouvedia** [comment]
Since css files can be concatenated and to know whether it is being reused in another file would require #4088, I think:

- it shouldn't be fixable (if we are using a variable)
- it should report correctly

i.e.
```
/* case "legacy" */
--foo-bar: 255 128 64; // do not fix
--baz-qux: 255, 128, 65; // do not fix

color: rgba(var(--foo-bar), 0.5); // report
color: rgba(var(--foo-bar) / 0.5); // report

color: rgba(var(--baz-qux), 0.5);
color: rgba(var(--baz-qux) / 0.5); // report
```

```
/* case "modern" */
--foo-bar: 255 128 64; // do not fix
--baz-qux: 255, 128, 65; // do not fix

color: rgba(var(--foo-bar), 0.5); // report
color: rgba(var(--foo-bar) / 0.5);

color: rgba(var(--baz-qux), 0.5); // report
color: rgba(var(--baz-qux) / 0.5); // report
```

**Comment by jeddy3** [comment]
As:

- [rules are strict by default, but can be made more permissive with `ignore` options](https://stylelint.io/developer-guide/rules#write-the-rule)
- we can't resolve custom properties (as @Mouvedia [mention](https://github.com/stylelint/stylelint/issues/6620#issuecomment-1422209200))

We can add an `ignore: ["with-var-inside"]` option.

If a person wants to use legacy notation, whether directly or indirectly via custom properties, they can use `"legacy"`:

```css
:root { --foo: 255, 128, 64; }
a { color: rgba(var(--foo), 0.5); }
```

If they want to use modern notation, whether directly or indirectly via custom properties, they can use `"modern"`:

```css
root { --foo: 255 128 64; }
a { color: rgb(var(--foo) / 0.5); }
```

If they are, for example, migrating a codebase from legacy to modern notation and still have some custom properties in legacy notation, they can use `["modern", { "ignore": ["with-var-inside"] }`] which will make the rule safe to use:

```css
root { 
  --foo: 255 128 64;
  --bar: 0, 0, 10;
}
a { color: rgb(var(--foo) / 0.5); }
a { color: rgba(var(--bar), 0.5); }
a { color: rgb(120 30 50 / 0.5); }
```

And if they want to enforce neither notation consistently, then can turn the rule off with `null`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.
